src/huggingface_hub/utils/_errors.py

```diff
@@ -0,0 +1,41 @@
+from requests import HTTPError
+
+
+class RepositoryNotFoundError(HTTPError):
+    """
+    Raised when trying to access a hf.co URL with an invalid repository name, or
+    with a private repo name the user does not have access to.
+    """
+
+
+class EntryNotFoundError(HTTPError):
```

Comment:
Should this maybe be `FileNotFoundError`?

Reply:
Will let @julien-c comment on the choice of "entry", but I'd rather stay as close as possible to the error thrown by the backend

Reply:
no strong opinion! up to you.

Maybe we can document that this mean that the repo AND the revision exist/are accessible, BUT not the file/entry? or is it clear enough already?